dll/win32/winmm/time.c

```diff
@@ -422,7 +423,15 @@ MMRESULT WINAPI timeGetDevCaps(LPTIMECAPS lpCaps, UINT wSize)
 MMRESULT WINAPI timeBeginPeriod(UINT wPeriod)
 {
     if (wPeriod < MMSYSTIME_MININTERVAL || wPeriod > MMSYSTIME_MAXINTERVAL)
-	return TIMERR_NOCANDO;
+		return TIMERR_NOCANDO;
+
+	/*  High resolution timer requested, use QPC */
+	if (wPeriod <= 5 && TIME_qpcFreq.QuadPart == 0) {
+		if(QueryPerformanceFrequency(&TIME_qpcFreq))
+			TIME_qpcFreq.QuadPart /= 1000;
```

Comment:
So the value calculated here doesn't depend on wPeriod. Should it?

Reply:
A timer value less than 5 milliseconds (the default on NT) is a request for a high-accuracy timer.  This commit is meant to be an improvement over simply wrapping GetTickCount.